### PR TITLE
refactor: vote message proto conversion and re-organise protos/conversions

### PR DESCRIPTION
**Pull request by sdbondi**
Description
---
- organise proto conversions into `common`, `consensus`, `network`, `rpc` and `transaction`

Motivation and Context
---

- `common` - types common to all protos
- `consensus` - used by hotstuff
- `transaction` - the transaction payload types
- `rpc` - used in p2p rpc
- `network` - DAN protocol message types

How Has This Been Tested?
---
Manaully

### CODE REVIEW

File: applications/tari_validator_node/proto/dan/rpc.proto [hunk 1]
```diff
@@ -21,22 +21,22 @@
...
 
 import "common.proto";
 import "consensus.proto";
 import "network.proto";
+import "transaction.proto";
 
-service ValidatorNode {
-  rpc SubmitTransaction(SubmitTransactionRequest) returns (SubmitTransactionResponse);
-}
+//service ValidatorNode {
```
Comment (SWvheerden) [inline applications/tari_validator_node/proto/dan/rpc.proto]: test code?

File: applications/tari_validator_node/src/p2p/proto/conversions/consensus.rs [hunk 2]
```diff
@@ -0,0 +1,432 @@
...
+        let hash = TreeNodeHash::try_from(node.hash)?;
+        let parent = TreeNodeHash::try_from(node.parent)?;
+        let height = node.height;
+        let is_committed = node.is_committed;
+
+        Ok(Self::new(hash, parent, height, is_committed))
+    }
+}
+
+// // -------------------------------- KeyValue -------------------------------- //
```
Comment (SWvheerden) [inline applications/tari_validator_node/src/p2p/proto/conversions/consensus.rs]: test code?

File: applications/tari_validator_node/src/p2p/proto/conversions/mod.rs [hunk 3]
```diff
@@ -0,0 +1,26 @@
+//  Copyright 2024, The Tari Project
```
Comment (SWvheerden) [inline applications/tari_validator_node/src/p2p/proto/conversions/mod.rs]:
```suggestion
//  Copyright 2022, The Tari Project
```

File: applications/tari_validator_node/src/p2p/proto/conversions/transaction.rs [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+        let meta = transaction.meta().clone();
+        let (instructions, signature, sender_public_key) = transaction.destruct();
+
+        proto::transaction::Transaction {
+            instructions: instructions.into_iter().map(Into::into).collect(),
+            signature: Some(signature.signature().into()),
+            sender_public_key: sender_public_key.to_vec(),
+            fee,
+            meta: Some(meta.into()),
+            // balance_proof: todo!(),
```
Comment (SWvheerden) [inline applications/tari_validator_node/src/p2p/proto/conversions/transaction.rs:80]: test?

File: applications/tari_validator_node/src/p2p/proto/conversions/transaction.rs [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+                    .map_err(|_| anyhow!("invalid component_address"))?;
+                let method = request.method;
+                tari_engine_types::instruction::Instruction::CallMethod {
+                    template_address,
+                    component_address,
+                    method,
+                    args,
+                }
+            },
+            // 2 => tari_dan_engine::instruction::Instruction::PutLastInstructionOutputOnWorkspace { key: request.key },
```
Comment (SWvheerden) [inline applications/tari_validator_node/src/p2p/proto/conversions/transaction.rs:124]:
```suggestion
           
```

**Review by SWvheerden — COMMENTED** [review]
looks good,

It just has some uncommented code in it.

**Review by stringhandler — APPROVED** [review]

**Pull request merged**